### PR TITLE
Devex entity validation

### PR DESCRIPTION
**Pull request by matthopson**
First pass at adding `joi` validation for all properties on entities. In most cases, the additions were `optional` (as that was the easiest first step). Please keep an eye out for things that should be locked down a little tighter.

### CODE REVIEW

**Review by rachaelparris — APPROVED** [review]

File: shared/src/business/entities/cases/PublicDocketRecordEntry.js [hunk 1]
```diff
@@ -17,6 +17,27 @@ function PublicDocketRecordEntry(rawDocketEntry) {
   this.filingDate = rawDocketEntry.filingDate;
 }
 
-joiValidationDecorator(PublicDocketRecordEntry, joi.object(), undefined, {});
+joiValidationDecorator(
+  PublicDocketRecordEntry,
+  joi.object().keys({
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/PublicDocketRecordEntry.js:22]: I've wondered whether the `joiValidationDecorator` should compare the output of `Object.getOwnPropertyNames(new PublicDocketRecordEntry({}))` and compare keys against those provided to joi validation.

**Review by kkoskelin — APPROVED** [review]
Cool, man.

File: shared/src/business/entities/cases/PublicDocketRecordEntry.js (hunk 1 above)
Comment (kkoskelin) [reply shared/src/business/entities/cases/PublicDocketRecordEntry.js:22]: This is not something you need to do, or is necessarily even a good idea.  This is just me pondering aloud. do with it what you will :)

Comment (matthopson) [reply shared/src/business/entities/cases/PublicDocketRecordEntry.js:22]: It'd be a good way to setup some warnings for fields missing in validation. 🤔We do have some instances of object mutation (like in the case of formattedCaseDetail where that _could_ produce unexpected (or maybe they'd be _expected_) results.

**Pull request merged**